cmd/root.go

```diff
@@ -58,18 +85,27 @@ var rootCmd = &cobra.Command{
 
 		select {
 		case <-quit:
-			err := a.Shutdown(context.TODO())
+			err := adapter.Shutdown(context.TODO())
 			if err != nil {
-				opts.Logger.Error("An error occurred while exiting.", zap.Error(err))
+				logger.Error("An error occurred while exiting: ", err)
 				return
 			}
-			opts.Logger.Info("See you next time!")
+			err = logger.Sync()
+			if err != nil {
+				logger.Error("An error occurred while exiting: ", err)
+				return
+			}
+			logger.Info("See you next time!")
 		}
 	},
 }
 
 // Execute bootstrap root command.
 func Execute() {
+	// declare flags
+	rootCmd.PersistentFlags().StringVarP(&configFile, "config", "c", "", "config file")
```

Comment:
We can also add the logging options.

Reply:
@tokers I found that the `api7/gopkg` log module we are using does not support setting the log level dynamically, so we cannot read the configuration after initializing the logger to set the log level. And allowing users to set log level in the CLI doesn't feel good, we even have to add parameters like log file store path. 🤔

Reply:
A lot of programs support setting log level in the CLI, like ETCD, Ingress Controller and etc.

And the logger definitely supports setting the level dynamically. https://pkg.go.dev/github.com/api7/gopkg@v0.1.2/pkg/log#WithLogLevel

Reply:
> And the logger definitely supports setting the level dynamically. 

Yes, I noticed that, but it looks like this won't work after the logger has been created. 

The current boot process is as follows.

```text
 init logger (setup an default level logger) => init config (re-config log level and more, like log file path, but not supported; there are some log output in this stage) => start etcd-adapter
```

Currently we can define this level parameter via environment variables, but that needs to be done after parsing the configuration file normally (we can't reset the level at that time). If you think you do need to support configuring this property via the CLI, I will try to add a flag to config binding to make that.